### PR TITLE
Add nullchecks for scene

### PR DESCRIPTION
**Pull request by JShep1**
These nullchecks are needed for a mock custom render engine plugin example I'm currently writing [here](https://github.com/ignitionrobotics/ign-rendering/pull/154).

Signed-off-by: John Shepherd <john@openrobotics.org>

### CODE REVIEW

File: src/gui/plugins/scene3d/Scene3D.cc [hunk 1]
```diff
@@ -441,6 +441,9 @@ void IgnRenderer::Render()
   if (!this->dataPtr->followTarget.empty())
   {
     rendering::ScenePtr scene = this->dataPtr->renderUtil.Scene();
+    if (!scene)
```
Comment (chapulina) [inline src/gui/plugins/scene3d/Scene3D.cc:444]: How about getting the scene once at the beginning of `IgnRenderer::Render()` and printing a warning? This way, if we add other scene calls to the body of this function, we don't need to remember to check for `nullptr`.

Comment (JShep1) [reply src/gui/plugins/scene3d/Scene3D.cc]: Fixed here 3008548703b50ab7916bcba68e047c4bd5c456c8 .  I might need some help on the warning message, wasn't sure what exactly to put

Comment (chapulina) [reply src/gui/plugins/scene3d/Scene3D.cc]: That looks good to me

**Review by chapulina — APPROVED** [review]
LGTM, let's wait for Jenkins to come back online to check CI

**Comment by JShep1** [comment]
> LGTM, let's wait for Jenkins to come back online to check CI

:+1: I also retargeted at another branch of mine adding support for custom render engine to make the dependency easier to pull when testing the `ign-rendering` functionality

**Pull request merged**